qiskit_research/utils/pulse_scaling.py

```diff
@@ -94,7 +95,28 @@ def run(self, dag: DAGCircuit) -> DAGCircuit:
                     if len(chunk) > 1:
                         for node in chunk[1:]:
                             dag.remove_op_node(node)
-            return dag
+        return dag
+
+
+class ReduceAngles(TransformationPass):
+    """Reduce angle of scaled pulses to between -pi and pi.
+
+    This works only after Parameters are bound. Gate strings
+    should only be single-parameter scaled pulses, i.e.
+    'rzx' and 'secr'.
+    """
+
+    def __init__(self, gate_strs: List[str]):
+        super().__init__()
+        self._gate_strs = gate_strs
+
+    def run(self, dag: DAGCircuit) -> DAGCircuit:
+        for gate_str in self._gate_strs:
+            for grun in dag.collect_runs([gate_str]):
+                theta = grun[0].op.params[0]
+                grun[0].op.params[0] = (float(theta) + pi) % (2 * pi) - pi
```

Comment:
Doesn't this only modify the first gate of the run? Also, please rename `grun` to simply `run`.